ibm/resource_ibm_cr_namespace.go

```diff
@@ -1,214 +1,234 @@
+				Deprecated:  "This field is deprecated",
+			},
+			"updated_on": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "When the namespace was last updated.",
+				Deprecated:  "This field is deprecated",
 			},
 		},
 	}
 }
-func resourceIBMCrNamespaceValidator() *ResourceValidator {
 
+func resourceIBMCrNamespaceValidator() *ResourceValidator {
 	validateSchema := make([]ValidateSchema, 1)
-
 	validateSchema = append(validateSchema,
 		ValidateSchema{
 			Identifier:                 "name",
 			ValidateFunctionIdentifier: ValidateRegexpLen,
 			Type:                       TypeString,
 			Required:                   true,
-			Regexp:                     `^[a-z0-9]+[a-z0-9\-\_]+[a-z0-9]+$`,
+			Regexp:                     `^[a-z0-9]+[a-z0-9_-]+[a-z0-9]+$`,
 			MinValueLength:             4,
-			MaxValueLength:             30})
+			MaxValueLength:             30,
+		},
+	)
 
-	ibmCrNamespaceResourceValidator := ResourceValidator{ResourceName: "ibm_cr_namespace", Schema: validateSchema}
-	return &ibmCrNamespaceResourceValidator
+	resourceValidator := ResourceValidator{ResourceName: "ibm_cr_namespace", Schema: validateSchema}
+	return &resourceValidator
 }
-func resourceIBMContainerRegistryNamespaceCreate(d *schema.ResourceData, meta interface{}) error {
-	userDetails, err := meta.(ClientSession).BluemixUserDetails()
-	if err != nil {
-		return err
-	}
-	accountID := userDetails.userAccount
-	log.Print("accountID", accountID)
 
-	crClient, err := meta.(ClientSession).ContainerRegistryAPI()
+func resourceIBMCrNamespaceCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	containerRegistryClient, err := meta.(ClientSession).ContainerRegistryV1()
 	if err != nil {
-		return err
-	}
-	crAPI := crClient.Namespaces()
-	namespace := d.Get("name").(string)
-	target := registryv1.NamespaceTargetHeader{
-		AccountID: accountID,
+		return diag.FromErr(err)
 	}
-	if rg, ok := d.GetOk("resource_group_id"); ok {
-		target.ResourceGroup = rg.(string)
+
+	createNamespaceOptions := &containerregistryv1.CreateNamespaceOptions{}
+
+	createNamespaceOptions.SetName(d.Get("name").(string))
+	if _, ok := d.GetOk("resource_group_id"); ok {
+		createNamespaceOptions.SetXAuthResourceGroup(d.Get("resource_group_id").(string))
 	} else {
 		defaultRg, err := defaultResourceGroup(meta)
 		if err != nil {
-			return err
+			return diag.FromErr(err)
 		}
-		target.ResourceGroup = defaultRg
+		createNamespaceOptions.SetXAuthResourceGroup(defaultRg)
 	}
 
-	response, err := crAPI.AddNamespace(namespace, target)
+	namespace, response, err := containerRegistryClient.CreateNamespaceWithContext(context, createNamespaceOptions)
 	if err != nil {
-		return err
+		log.Printf("[DEBUG] CreateNamespaceWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
 	}
-	d.SetId(response.Namespace)
-	return resourceIBMContainerRegistryNamespaceRead(d, meta)
+
+	d.SetId(*namespace.Namespace)
+
```

Comment:
@hkantare Can you please open an issue on the generator for this feature?  Currently the generator only supports "local tags", but I agree that we should use ghost where appropriate.

Reply:
Sadly, this turns out to be difficult for namespaces, since the response to a namespace create _does not_ provide the CRN of the namespace.  So before I spend time working out the hand-edits that are needed to make this work for namespaces, I want to ask how important it is to have "formal" tags for namespaces instead of "local" tags.

Reply:
I will further note that the UI for namespaces does not show tags associated with a namespace (as best I can tell), and namespaces do not show up in the "Resource list", so I don't see any way to verify that tags were applied to the namespace if indeed we can figure out how to do it.

Reply:
Agree with Mike here.. There is further context (but for a different forum).

Reply:
Ok I agree with Mike and James...you can for now ignore tags